### PR TITLE
feat(rule): add popup before leave rule adding/editing page

### PR DESCRIPTION
**Pull request by Kinplemelon**

### CODE REVIEW

**Comment by Kinplemelon** [comment]
image

File: src/hooks/Rule/rule/useRuleEditingPageUnload.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    } catch (error) {
+      return false
+    }
+  })
+
+  onMounted(bindEventListener)
+
+  onUnmounted(unbindEventListener)
+
+  return {}
```
Comment (ysfscream) [inline src/hooks/Rule/rule/useRuleEditingPageUnload.ts:64]: Why return empty object?

File: src/hooks/Rule/rule/useRuleEditingPageUnload.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  }
+
+  const unbindEventListener = () => {
+    window.removeEventListener('beforeunload', unloadHandler)
+  }
+
+  const customPopupWarning = () => {
+    if (getters.isDev) {
+      return Promise.resolve()
+    }
```
Comment (ysfscream) [inline src/hooks/Rule/rule/useRuleEditingPageUnload.ts:34]: Why we use the isDev?

File: src/common/auth.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  const ek = getEncryptionKey()
+  if (!uk || !ek) {
+    return
+  }
+  const user = localStorage.getItem(uk)
+  if (!user) {
+    return
+  }
+  const userInfoStr = decryptSafely(user, ek) as string
+  console.log(userInfoStr)
```
Comment (ysfscream) [inline src/common/auth.ts:74]: remove this console

File: src/hooks/Rule/rule/useRuleEditingPageUnload.ts (hunk 1 above)
Comment (Kinplemelon) [reply src/hooks/Rule/rule/useRuleEditingPageUnload.ts]: fixed

File: src/common/auth.ts (hunk 3 above)
Comment (Kinplemelon) [reply src/common/auth.ts]: removed

**Review by ysfscream — APPROVED** [review]

**Pull request merged**